lib/internal/per_context/primordials.js

```diff
@@ -99,28 +114,28 @@ const createSafeIterator = (factory, next) => {
     next() {
       return next(this._iterator);
     }
-    [Symbol.iterator]() {
+    [SymbolIterator]() {
       return this;
     }
   }
-  Object.setPrototypeOf(SafeIterator.prototype, null);
-  Object.freeze(SafeIterator.prototype);
-  Object.freeze(SafeIterator);
+  ObjectSetPrototypeOf(SafeIterator.prototype, null);
+  ObjectFreeze(SafeIterator.prototype);
+  ObjectFreeze(SafeIterator);
   return SafeIterator;
 };
 
 function makeSafe(unsafe, safe) {
-  if (Symbol.iterator in unsafe.prototype) {
+  if (SymbolIterator in unsafe.prototype) {
     const dummy = new unsafe();
     let next; // We can reuse the same `next` method.
 
-    for (const key of Reflect.ownKeys(unsafe.prototype)) {
-      if (!Reflect.getOwnPropertyDescriptor(safe.prototype, key)) {
-        const desc = Reflect.getOwnPropertyDescriptor(unsafe.prototype, key);
+    for (const key of ReflectOwnKeys(unsafe.prototype)) {
+      if (!ReflectGetOwnPropertyDescriptor(safe.prototype, key)) {
+        const desc = ReflectGetOwnPropertyDescriptor(unsafe.prototype, key);
         if (
           typeof desc.value === 'function' &&
           desc.value.length === 0 &&
-          Symbol.iterator in (desc.value.call(dummy) ?? {})
+          SymbolIterator in (desc.value.call(dummy) ?? {})
```

Comment:
If we're gonna primordials, let's primordials all the way ;)
```suggestion
          SymbolIterator in (call(desc.value, dummy) ?? {})
```

Reply:
`call` isn’t uncurried though.